wagtail/core/models.py

```diff
@@ -87,6 +87,8 @@ def _get_field_dictionaries(self, exclude_fields=None):
                             continue
                     except AttributeError:
                         pass
+
+                    # TODO: Wouldn't this reassign the objects to the new page rather than copy them?
```

Comment:
@kaedroho Isn't that what you want to happen for an M2M relation? If a page has an M2M relation linking it to several categories, then you want the copied page to be linked to those same categories - you don't want it to duplicate the category objects.

Reply:
I was just wondering if this would remove them from the source page if we were passing in the through model instances here?

Reply:
I don't think so - on the line below we're calling `all()` on the relation, which skips over the through model and returns a queryset of the model at the far end of the relation. When these are assigned back to the relation on the copied page, it will create new through models for them.

That does have me wondering what should happen if there's an explicit through model with extra fields, though: as it stands, that extra data will be lost on the copied page (or fail if any of those fields are required). In fact, at this point I'm not even sure what the proper use-case is for a non-parental ManyToMany field on a ClusterableModel, and whether it's really appropriate to copy these relations across at all. There's the one special case being addressed directly above, where an M2M relation is being set up as a convenience for accessing a classic ParentalKey/InlinePanel relation ("get all of this image gallery's Image objects"), which makes sense - but other than that, it seems to me that non-parental ManyToMany relations aren't going to be used for actual page content, and their usefulness is confined to things like "user X has bookmarked page Y". And in that case, you really don't want the relation to be copied...

Reply:
Given this is a refactoring PR, shall we open a new PR to change that behaviour?

Reply:
Sure, that sounds fair (and if non-parental M2Ms aren't really used in practice, maybe this isn't a priority to fix anyhow). This comment stopped me in my tracks because it looked like you might have gone on to wrongly fix this by duplicating the objects on the other side of the relation... if that's not the case, then I'm happy to continue reviewing!